client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -170,23 +177,24 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerI
         long traceId = LoggerHelpers.traceEnter(log, "createScope", scopeName);
 
         final CompletableFuture<CreateScopeStatus> result = this.retryConfig.runAsync(() -> {
-                RPCAsyncCallback<CreateScopeStatus> callback = new RPCAsyncCallback<>(traceId, "createScope");
-                client.createScope(ScopeInfo.newBuilder().setScope(scopeName).build(), callback);
-                return callback.getFuture();
+            RPCAsyncCallback<CreateScopeStatus> callback = new RPCAsyncCallback<>(traceId, "createScope");
+            addTagsToRequest(client, traceId, "createScope", scopeName)
+                    .createScope(ScopeInfo.newBuilder().setScope(scopeName).build(), callback);
+            return callback.getFuture();
         }, this.executor);
         return result.thenApply(x -> {
                 switch (x.getStatus()) {
                 case FAILURE:
-                    log.warn("Failed to create scope: {}", scopeName);
+                    LoggerHelpers.warnLogWithTag(log, traceId, "Failed to create scope: {}", scopeName);
                     throw new ControllerFailureException("Failed to create scope: " + scopeName);
                 case INVALID_SCOPE_NAME:
-                    log.warn("Illegal scope name: {}", scopeName);
+                    LoggerHelpers.warnLogWithTag(log, traceId, "Illegal scope name: {}", scopeName);
```

Comment:
Instead of replacing all the `log.*` calls with `LoggerHelpers.warnLogWithTag`, I think we can do something cute with inheriting from `org.slf4j.Logger` through `@Slf4j` and/or use Slf4J markers (https://www.slf4j.org/api/org/slf4j/Marker.html).
Have you thought about this? I think it will modularize the changes.

Reply:
Please, see the last comment about `TagLogger`.